File: pset6/readability/readability.py
from sys import argv, exit

def main():
    """Implementation with text provided as an argument
    # Check if text is provided.
    if len(argv) != 2:
        print("Please provide a text as an argument.")
        exit()
    """

    # Get counters.
    letters, words, sentences = count(get_text())
    l = letters * (100 / words)
    s = sentences * (100 / words)
    # Get index.
    grade = col(l, s)
    print_result(grade)

# Function for countnig letters, words and sentences.
def count(text):
    letters = 0
    words = 0
    sentences = 0
    for i in text:
        if i.isalpha():
            letters += 1
        elif i == " ":
            words += 1
        elif i in ['.', '!', '?']:
            sentences += 1
        else:
            continue
    words += 1  # Add the last word from text.

    return letters, words, sentences

# Function to calculate Coleman-Liau index.
def col(l, s):
    return round(0.0588 * l - 0.296 * s - 15.8)

# Get text function.
def get_text():
    text = input("Enter text for evaluation: ")
    return text

# Print result function.
def print_result(grade):
    if grade < 1:
        print("Before grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

void get_input(char text[]);
void counter(char text[], int *words, int *letters, int *sentences);
float index(int letters, int words, int sentences);

int main(void){
    // Variables for storing text, word, letter and sentence count.
    char text[1000];
    int words = 0;
    int letters = 0;
    int sentences = 0;
    int grade;

    // Get user input.
    printf("Text:\n");
    get_input(text);

    // Count letters, words and sentences.
    counter(text, &words, &letters, &sentences);

    // Evaluate grade.
    grade = index(letters, words, sentences);
    if (grade < 1) {
        printf("Before Grade 1\n");
    }
    else if (grade > 16) {
        printf("Grade 16+\n");
    }
    else {
        printf("Grade %i\n", grade);
    }

    return 0;
}

// Store user input.
void get_input(char text[]) {
    scanf("%[^\n]s", text);
}

// Counter function.
void counter(char text[], int *words, int *letters, int *sentences) {
    for (int i = 0; i < strlen(text); i++) {
        if (isalpha(text[i])) {
            *letters += 1;
        }
        else if (text[i] == ' ') {
            *words += 1;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?') {
            *sentences += 1;
        }
    }
    *words += 1;
}

// Define index.
float index(int letters, int words, int sentences) {
    float i = 0.0588 * (100 / (float)words) * letters - 0.296 * (100 / (float)words) * sentences - 15.8;
    return round(i);
}